nuxeo-services/nuxeo-platform-directory/nuxeo-platform-directory-api/src/main/java/org/nuxeo/ecm/directory/Directory.java

```diff
@@ -38,6 +38,27 @@
 // TODO: maybe separate Directory implementation and Directory Source
 public interface Directory {
 
+    /**
+     * INTERNAL, DO NOT CALL. Initializes the directory when Nuxeo starts. Called without a transaction.
```

Comment:
Would there be a pattern to avoid this? Maybe with the next versions of Java?

Reply:
One possible pattern would be similar to what I did for `KeyValueStore` / `KeyValueStoreProvider` but I didn't want to split a separate `DirectoryProvider` API in this ticket.